src/Magnum/DebugTools/BufferData.h

```diff
@@ -47,35 +47,45 @@ namespace Implementation {
 @brief Buffer subdata
 
 Emulates @ref GL::Buffer::subData() call on platforms that don't support it
-(such as OpenGL ES) by using @ref GL::Buffer::map().
+(such as OpenGL ES) by using @ref GL::Buffer::mapRead(). On desktop GL and
+WebGL 2.0 it's just an alias to @ref GL::Buffer::subData().
 
 @note This function is available only if Magnum is compiled with
     @ref MAGNUM_TARGET_GL "TARGET_GL" enabled (done by default). See
     @ref building-features for more information.
 
 @requires_gles30 Extension @gl_extension{EXT,map_buffer_range} in OpenGL ES
     2.0.
-@requires_gles Buffer mapping is not available in WebGL.
+@requires_webgl20 Buffer data queries or buffer mapping are not available in
+    WebGL 1.0.
 */
-template<class T> Containers::Array<T> inline bufferSubData(GL::Buffer& buffer, GLintptr offset, GLsizeiptr size) {
+template<class T = char> Containers::Array<T> inline bufferSubData(GL::Buffer& buffer, GLintptr offset, GLsizeiptr size) {
+    if(!size) return {};
+    #if !defined(MAGNUM_TARGET_GLES) || defined(MAGNUM_TARGET_WEBGL)
+    Containers::Array<char> data = buffer.subData(offset, size*sizeof(T));
+    return Containers::Array<T>{reinterpret_cast<T*>(data.release()), std::size_t(size)};
```

Comment:
This code assumes that `GL::Buffer::subData()` always returns an `Array` with a default deleter. Is that a valid assumption? If so, should I assert for it here?

Reply:
Good point -- yeah an `CORRADE_INTERNAL_ASSERT(!data.deleter());` would be great.

Hmm, thinking about this, it'd probably make more sense if I dropped the templated aspect of this. `Buffer::data()` used to have that as well, but it was *so* error-prone ("what is the size? bytes? elements??") that I eventually deprecated and removed that, telling people to do an `arrayCast()` instead. I'll add that as a post-merge TODO.

Reply:
The template parameter is pretty handy but I can see why it's confusing. I've already been using it in "[GL, Text: verify buffer contents on GLES3](https://github.com/mosra/magnum/pull/560/commits/9fa66f00cebe0a3b8a1d503e396dcd012f1e7729)", should I change those already or leave it for post-merge too?